compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -517,7 +517,13 @@ class ComposeScene internal constructor(
      * Render the current content on [canvas]. Passed [nanoTime] will be used to drive all
      * animations in the content (or any other code, which uses [withFrameNanos]
      */
-    fun render(canvas: Canvas, nanoTime: Long): Unit = postponeInvalidation {
+    fun render(canvas: Canvas, nanoTime: Long): Unit = render(retrieveCanvas = { canvas }, nanoTime)
+
+    /**
+     * Flush async operations, notify all animations and perform recomposition-layout-draw sequence.
+     * [Canvas] to draw on is retrieved lazily to postpone acquiring drawable as late as possible.
+     */
+    fun render(retrieveCanvas: () -> Canvas?, nanoTime: Long): Unit = postponeInvalidation {
```

Comment:
It's a new public API.
1. Should we make it experimental at first?
2. Should we deprecate old one?

Reply:
Alternative solutions:
- separate to `doFrame` and `draw`
- make custom LazyCanvas implementation

Reply:
I moved it to internal temporarily. Public API doesn't change.
The old one should probably be deprecated. It's correct, but most likely unoptimal in terms of swap chain usage. Needs investigation on DirectX and OpenGL to prove or disprove it. Added TODO.